homeassistant/components/niko_home_control/config_flow.py

```diff
@@ -0,0 +1,75 @@
+"""Config flow for the Niko home control integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from nikohomecontrol import NikoHomeControlConnection
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+from .const import DEFAULT_IP, DEFAULT_PORT, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_IP): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+    }
+)
+
+IDENTIFIER = f"${DOMAIN}.controller"  # only one controller is supported
+
+
+def test_connection(host: str, port: int) -> str | None:
+    """Test if we can connect to the Niko Home Control controller."""
+    try:
+        if NikoHomeControlConnection(host, port):
+            return None
+    except Exception:  # noqa: BLE001
+        return "unknown"
+    else:
+        return "cannot_connect"
+
+
+class NikoHomeControlConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Niko Home Control."""
+
+    VERSION = 1
+
+    def __init__(self, import_info: dict[str, str] | None = None) -> None:
+        """Initialize the config flow."""
+        self._import_info = import_info
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            error = test_connection(user_input[CONF_HOST], user_input[CONF_PORT])
+            if not error:
+                await self.async_set_unique_id(IDENTIFIER)
+                self._abort_if_unique_id_configured()
```

Comment:
https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/unique-config-entry#unique-data

Rather check on unique data, so it is possible to set up mutliple

Reply:
The thing is that it is not advised to have multiple controllers in one house.

Reply:
The thing with HA is that people have weird setups and edge cases, so there will be people who run that. So if its technical possible, I would always try to allow it, just for the sake of consistency

Reply:
It's one controller for one house if there would be the need to control another nhc controller the user would be better of using another HA setup.

Reply:
Yes, but it won't be the first time people have their vacation home connected via VPN to only use one HA instance. I am just throwing ideas now, but if we can support it and it is no hassle, it's most likely little to no extra work to support

Reply:
Well, I have clients like that but they have allot of local stuff so need HA running in the vacation home

Reply:
But that's not the point. The point is, we can add a little code to support it and never care about it again :)

Reply:
I agree, was working on it yesterday but then came the link issue in docs😂